pkg/lib/bundle/validate_test.go

```diff
@@ -73,10 +73,74 @@ func TestValidateBundle_InvalidRegistryVersion(t *testing.T) {
 		logger: logger,
 	}
 
-	err := validator.ValidateBundle(dir)
+	err := validator.ValidateBundleFormat(dir)
 	require.Error(t, err)
 	var validationError ValidationError
 	isValidationErr := errors.As(err, &validationError)
 	require.True(t, isValidationErr)
-	require.Equal(t, len(validationError.AnnotationErrors), 1)
+	require.Equal(t, len(validationError.Errors), 1)
+}
+
+func TestValidateBundleContent(t *testing.T) {
+	logger := logrus.NewEntry(logrus.New())
+
+	validator := imageValidator{
+		logger: logger,
+	}
+
+	var table = []struct {
+		description string
+		mediaType   string
+		directory   string
+		numErrors   int
+		errString   string
+	}{
+		{
+			description: "registryv1 bundle/invalid csv",
+			mediaType:   RegistryV1Type,
+			directory:   "./testdata/validate/invalid_manifests_bundle/invalid_csv/",
+			numErrors:   1,
+			errString:   "install modes not found",
+		},
+		{
+			description: "registryv1 bundle/invalid crd",
+			mediaType:   RegistryV1Type,
+			directory:   "./testdata/validate/invalid_manifests_bundle/invalid_crd/",
+			numErrors:   1,
+			errString:   "cannot restore slice from string",
+		},
+		{
+			description: "registryv1 bundle/invalid sa",
+			mediaType:   RegistryV1Type,
+			directory:   "./testdata/validate/invalid_manifests_bundle/invalid_sa/",
+			numErrors:   1,
+			errString:   "json: cannot unmarshal number into Go struct field ObjectMeta.metadata.namespace of type string",
+		},
+		{
+			description: "registryv1 bundle/invalid type",
+			mediaType:   RegistryV1Type,
+			directory:   "./testdata/validate/invalid_manifests_bundle/invalid_type/",
+			numErrors:   1,
+			errString:   "ResourceQuota is not supported type for registryV1 bundle",
+		},
+		{
+			description: "valid registryv1 bundle",
+			mediaType:   RegistryV1Type,
+			directory:   "./testdata/validate/valid_bundle/manifests/",
+			numErrors:   0,
+		},
+	}
+
+	for i, tt := range table {
+		fmt.Println(tt.directory)
+		err := validator.ValidateBundleContent(tt.directory)
+		var validationError ValidationError
+		isValidationErr := errors.As(err, &validationError)
+		require.True(t, isValidationErr)
+		require.Len(t, validationError.Errors, tt.numErrors, table[i].description)
+		if len(validationError.Errors) > 0 {
+			e := validationError.Errors[0]
+			require.Contains(t, e.Error(), tt.errString)
+		}
+	}
 }
```

Comment:
I can't comment on the files - but we should `.gitignore` the ` pkg/sqlite/migrations/1048603298946505393.db-journal` files